### PR TITLE
Send events to intercom

### PR DESCRIPTION
**Pull request by Ark2307**

### CODE REVIEW

File: apps/dashboard/src/main/java/com/akto/action/EventMetricsAction.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        if(shouldSendEvent){
+            try {
+                BasicDBList allUsers = UsersDao.instance.getAllUsersInfoForTheAccount(Context.accountId.get());
+                for(Object obj: allUsers) {
+                    BasicDBObject userObj = (BasicDBObject) obj;
+                    String userEmail = userObj.getString("login");
+                    String name = userObj.getString("name");
+                    event.setEmail(userEmail);
+                    Event.create(event);
+                    currentEventsMetrics.setCreatedAt(timeNow);
```
Comment (avneesh-akto) [inline apps/dashboard/src/main/java/com/akto/action/EventMetricsAction.java:57]: remove this line

File: libs/dao/src/main/java/com/akto/dao/ApiInfoDao.java [hunk 2]
```diff
@@ -157,6 +163,60 @@ public static Float getRiskScore(ApiInfo apiInfo, boolean isSensitive, float ris
...
+                        apiInfoList.add(a);
+                    }
+                }
+                urlsToSearch.clear();
+            } 
+        }
+        return apiInfoList;
+    }
+
+    public void insertMetricsForIntercomEvent(EventsMetrics lastEventSent, EventsMetrics currentEvent){
```
Comment (avneesh-akto) [inline libs/dao/src/main/java/com/akto/dao/ApiInfoDao.java:204]: rename this

Comment (avneesh-akto) [reply libs/dao/src/main/java/com/akto/dao/ApiInfoDao.java:204]: dao should have only db calls.

File: libs/dao/src/main/java/com/akto/dao/notifications/EventsMetricsDao.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            currentEvent.setCount(currentEvent.getCount() + 1);
+            if(currentEvent.getCount() < 5){
+                List<String> newExamples = currentEvent.getExamples();
+                newExamples.addAll(apiExampleList);
+                currentEvent.setExamples(newExamples);
+            }
+        }
+        localPostureMap.put(key, currentEvent);
+    }
+
```
Comment (avneesh-akto) [inline libs/dao/src/main/java/com/akto/dao/notifications/EventsMetricsDao.java:65]: replace eventmetrics with just timestamp

File: libs/dao/src/main/java/com/akto/dao/notifications/EventsMetricsDao.java [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    public void sendPostureMapToIntercom(EventsMetrics lastEventSent, EventsMetrics currentEvent){
+        int lastEventSentEpoch = 0;
+        if(lastEventSent != null){
+            lastEventSentEpoch = lastEventSent.getCreatedAt();
+        }
+
+        Map<String, EventsExample> localPostureMap = new HashMap<>();
+
+        if(currentEvent.getApiSecurityPosture() != null && !currentEvent.getApiSecurityPosture().isEmpty()){
```
Comment (avneesh-akto) [inline libs/dao/src/main/java/com/akto/dao/notifications/EventsMetricsDao.java:74]: remove

File: apps/dashboard/pom.xml [hunk 5]
```diff
@@ -242,6 +242,11 @@
             <artifactId>commons-csv</artifactId>
             <version>1.9.0</version>
         </dependency>
+        <dependency>
```
Comment (avneesh-akto) [inline apps/dashboard/pom.xml:245]: remove this

File: apps/dashboard/src/main/java/com/akto/action/EventMetricsAction.java (hunk 1 above)
Comment (Ark2307) [reply apps/dashboard/src/main/java/com/akto/action/EventMetricsAction.java:57]: done

File: libs/dao/src/main/java/com/akto/dao/ApiInfoDao.java (hunk 2 above)
Comment (Ark2307) [reply libs/dao/src/main/java/com/akto/dao/ApiInfoDao.java:204]: done

File: libs/dao/src/main/java/com/akto/dao/notifications/EventsMetricsDao.java (hunk 3 above)
Comment (Ark2307) [reply libs/dao/src/main/java/com/akto/dao/notifications/EventsMetricsDao.java:65]: done

File: libs/dao/src/main/java/com/akto/dao/notifications/EventsMetricsDao.java (hunk 4 above)
Comment (Ark2307) [reply libs/dao/src/main/java/com/akto/dao/notifications/EventsMetricsDao.java:74]: done

File: apps/dashboard/pom.xml (hunk 5 above)
Comment (Ark2307) [reply apps/dashboard/pom.xml:245]: done

**Pull request merged**